javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -87,7 +87,10 @@ protected static void AddOptionalAndConstantVariables(JavaBlock function, Client
                     (parameterClientType instanceof ArrayType || parameterClientType instanceof ListType)) {
                 parameterWireType = ClassType.String;
             }
-            boolean alwaysNull = parameterWireType != parameterClientType && clientMethod.getOnlyRequiredParameters() && !parameter.getIsRequired();
+            // "addConstant" clause is a hack here.
+            // For SimpleAsyncRestResponse and PagingAsyncSinglePage, "alwaysNull" is set to true, to let "ConvertClientTypesToWireTypes" method set the optional parameter.
+            // For other ClientMethodType (which would have addConstant=false), optional parameter need to be initialized for calling method with full parameters.
+            boolean alwaysNull = addConstant && parameterWireType != parameterClientType && clientMethod.getOnlyRequiredParameters() && !parameter.getIsRequired();
```

Comment:
So far, it is safe to 1. set the parameter to `null`, 2. convert the client type of `null` value to wire type.

Checked that following will all convert `null` to `null`, as intended.
```
Base64Util.encodeToString
Base64Url.encode
JacksonAdapter.createDefaultSerializerAdapter().serializeList
```

Reply:
I am not sure what `addConstant` is used for in other places but taking a dependency on this field to determine whether to init the optional param might cause issues when the semantics of `addConstant` change. Is there a more reliable way to determine whether optional param should be initialized or not?

Reply:
OK. Added `ignoreParameterNeedConvert` for this purpose.